legacy/src/main/java/org/opensearch/sql/legacy/cursor/DefaultCursor.java

```diff
@@ -132,8 +222,20 @@ private JSONObject schemaEntry(String name, String alias, String type) {
     return entry;
   }
 
-  private static String encodeCursor(JSONObject cursorJson) {
-    return Base64.getEncoder().encodeToString(cursorJson.toString().getBytes());
+  private static String encodeCursor(JSONObject cursorJson, SearchSourceBuilder sourceBuilder) {
+    try {
+      ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+      XContentBuilder builder = XContentFactory.jsonBuilder(outputStream);
+      sourceBuilder.toXContent(builder, null);
+      builder.close();
+
+      String searchRequestBase64 = Base64.getEncoder().encodeToString(outputStream.toByteArray());
+      cursorJson.put("searchSourceBuilder", searchRequestBase64);
```

Comment:
Is `searchSourceBuilder` the right naming? Also unsure if it is right to serialize the searchSourceBuilder and make it cursor. (builder didn't sound like a persisted object, but it is fine if this is a common way in OpenSearch)

Reply:
@manasvinibs  Please don't change anything in encodeCursor..before inovking this method...add 
`
 cursorJson.put("searchSourceBuilder", searchRequestBase64);
 `
 Add the above stuff in the invoker.

Reply:
> Is `searchSourceBuilder` the right naming? Also unsure if it is right to serialize the searchSourceBuilder and make it cursor. (builder didn't sound like a persisted object, but it is fine if this is a common way in OpenSearch)
I think Rupal spent sometime on this and after discussing with team came up with this approach. (Now that we need sort info for PIT search_after). I'm not sure how this can be improved, if you think its worth it can be something to take as follow-up.